issue1-seed-identification-api-for-nachet-frontend.py

```diff
@@ -0,0 +1,147 @@
+    7. Processes the response, extracting the name and saving it as a JSON file.
+    """
+    for url, seed_name in url_to_seed_mapping.items():
+        logging.info("Current seed: %s", seed_name)
+
+        seed_json_path = seed_name + ".json"
+
+        file_path = os.path.join(seed_data_path, seed_json_path)
+
+        if os.path.exists(file_path):
+            print(f"JSON file {seed_json_path} exists in {seed_data_path}, skipping")
+        else:
+            web_pages = query_get_webpage(cursor, url)
+
+            all_language_seed_page = ""
+            for row in web_pages:
+                web_text = row.get("cleaned_content")
+                all_language_seed_page += web_text
+            page = all_language_seed_page
+            md5hash = row.get("md5hash")
+
+            ### Get the images corresponding to the current page
+            images_fetch = query_get_images(cursor, md5hash)
+
+            image_information = ""
+
+            for row in images_fetch:
+                image_links = row["photo_link"]
+                image_descriptions = row["photo_description"]
+                image_information += f"Image link: {image_links}"
+                image_information += f"\nImage description: {image_descriptions}\n\n"
+
+            print("Sending request for summary to Azure OpenAI endpoint...\n")
+
+            user_prompt = (
+                load_user_prompt
+                + "Return a JSON file that follows this template:\n\n"
+                + json_template
+                + "\n\nhere is the text to parse:\n"
+                + page
+                + "\n\nAnd here is the images descriptions:\n"
+                + image_information
+            )
+
+            response = openai.get_chat_answer(system_prompt, user_prompt, 2000)
+
+            data = json.loads(response.choices[0].message.content)
+
+            if isinstance(data, dict):
+                file_name = seed_name
+                file_name = file_name.encode("latin1").decode("unicode-escape")
+                file_name += ".json"
+
+                file_path = os.path.join(SEED_DATA_PATH, file_name)
+                with open(file_path, "w") as json_file:
+                    json.dump(data, json_file, ensure_ascii=False, indent=4)
+
+                print(f"JSON data written to {file_path}")
+            else:
+                print("Error: not a dictionary, so it cannot be serialized to JSON.")
+
+
+if __name__ == "__main__":
+    CURRENT_WORKING_DIRECTORY = os.getcwd()
+    SEED_DATA_PATH = CURRENT_WORKING_DIRECTORY + "/seed-data"
+    PROMPT_PATH = CURRENT_WORKING_DIRECTORY + "/nachet-data/prompt"
+
+    system_prompt = nachet.load_prompt("system_prompt.txt", PROMPT_PATH)
+    load_user_prompt = nachet.load_prompt("user_prompt.txt", PROMPT_PATH)
+    json_template = nachet.load_json_template(PROMPT_PATH)
+
+    nachet_db = db.connect_db()
+    with nachet_db.cursor() as cursor:
+        seed_urls = query_seeds_urls(cursor, 3)
```

Comment:
If seed_urls are not used anywhere else, they might as well be created inside the `create_seed_url_mapping` function.

Reply:
This is a separate query, and I believe it's a better approach to maintain each query as distinct entities.